src/database/mod.rs

```diff
@@ -229,6 +229,33 @@ fn db_delegate_vault(
     })
 }
 
+/// Mark a vault as needing to be canceled
+pub fn db_should_cancel_vault(db_path: &path::Path, vault_id: i64) -> Result<(), DatabaseError> {
+    db_exec(&db_path, |db_tx| {
+        db_tx.execute(
+            "UPDATE vaults SET should_cancel = 1 WHERE id = (?1)",
+            params![vault_id],
+        )?;
+
+        Ok(())
+    })
+}
+
+/// Mark a vault as needing to go through
```

Comment:
nit: 'needing to go through' is a bit vague. how about "mark as should not cancel"?

Reply:
Done